flux-middle/src/fhir.rs

```diff
@@ -120,7 +122,8 @@ pub struct AliasTy {
 #[derive(Default, Debug)]
 pub struct Map {
     generics: FxHashMap<LocalDefId, Generics>,
-    predicates: FxHashMap<LocalDefId, GenericPredicates>,
+    predicates: ItemPredicates,
+    item_bounds: ItemPredicates,
```

Comment:
Do we need them both? Maybe we can store them both in the same map. Opaque types are also items that happen to have predicates and generics.